src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/ModelAttributes.cs

```diff
@@ -59,15 +76,22 @@ public ModelAttributes(IEnumerable<object> propertyAttributes, IEnumerable<objec
         public IReadOnlyList<object> Attributes { get; }
 
         /// <summary>
-        /// Gets the set of attributes on the property, or <c>null</c> if this instance represents the attributes
-        /// for a <see cref="Type"/>.
+        /// Gets the set of attributes on the property, or <c>null</c> if this instance does not represent the attributes
+        /// for a property.
         /// </summary>
         public IReadOnlyList<object> PropertyAttributes { get; }
 
         /// <summary>
+        /// Gets the set of attributes on the parameter, or <c>null</c> if this instance does not represent the attributes
+        /// for a parameter.
+        /// </summary>
+        public IReadOnlyList<object> ParameterAttributes { get; }
```

Comment:
PropertyAttributes and ParameterAttributes both serve the same purpose - they are a set of attributes that are *more specific* than the TypeAttributes. I'm not sure we need 3 things, when only 2 of them will be used at a time.

Should we introduce `MemberAttributes` as an alias for `ParameterAttributes` and then obsolete `ParameterAttributes`?

Reply:
Not sure about that. `ParameterAttributes` doesn't come with associated `TypeAttributes`, whereas `PropertyAttributes` always does. Plus `MemberAttributes` would be an odd name because [a parameter isn't a member in reflection terminology](https://msdn.microsoft.com/en-us/library/system.reflection.membertypes(v=vs.110).aspx). I think it will be clearer to have the three different slots that can be used for the three different types of thing rather than overloading one of them (even if two of the slots kind of could be munged together at a stretch).

Reply:
yeah that's true, ok